ui/src/conferences/containers/DetailPageContainer.jsx

```diff
@@ -1,11 +1,147 @@
+import RichDescription from '../../common/components/RichDescription';
+import ConferenceSeries from '../components/ConferenceSeries';
+import ContactList from '../../common/components/ContactList';
+import PublicNotesList from '../../common/components/PublicNotesList';
+import KeywordList from '../../common/components/KeywordList';
+import EditRecordAction from '../../common/components/EditRecordAction';
+import ProceedingsAction from '../components/ProceedingsAction';
+import ConferenceWebsitesAction from '../components/ConferenceWebsitesAction';
 
-function DetailPage() {
-  return <div>Detail</div>;
+function DetailPage({ loading, match, dispatch, record }) {
+  const recordId = match.params.id;
+
+  useEffect(
+    () => {
+      dispatch(fetchConference(recordId));
+    },
+    [dispatch, recordId]
+  );
+
+  const metadata = record.get('metadata');
+
+  if (!metadata) {
+    return null;
+  }
+
+  const title = metadata.getIn(['titles', 0]);
+  const acronym = metadata.getIn(['acronyms', 0]);
+  const openingDate = metadata.get('opening_date');
+  const closingDate = metadata.get('closing_date');
+  const location = metadata.getIn(['address', 0]);
+  const cnum = metadata.get('cnum');
+  const description = metadata.get('description');
+  const inspireCategories = metadata.get('inspire_categories');
+  const series = metadata.getIn(['series', 0]);
+  const contacts = metadata.get('contact_details');
+  const publicNotes = metadata.get('public_notes');
+  const keywords = metadata.get('keywords');
+  const urls = metadata.get('urls');
+  const proceedings = metadata.get('proceedings'); // TODO: check exact format after INSPIR-2850
+  const canEdit = metadata.get('can_edit', false);
+
+  return (
+    <>
+      <DocumentHead title={title.get('title')} />
+      <Row type="flex" justify="center">
+        <Col className="mv3" xs={24} md={22} lg={21} xxl={18}>
+          <ContentBox
+            loading={loading}
+            className="sm-pb3"
+            leftActions={
+              <>
+                {urls && <ConferenceWebsitesAction websites={urls} />}
+                {proceedings && <ProceedingsAction proceedings={proceedings} />}
+                {canEdit && (
+                  <EditRecordAction
+                    pidType="conferences"
+                    pidValue={this.recordId}
+                  />
+                )}
+              </>
+            }
+          >
+            <Row>
+              <Col>
+                <h2>
+                  <ConferenceTitle title={title} acronym={acronym} />
+                </h2>
+              </Col>
+            </Row>
+            <Row>
+              <Col>
+                <ConferenceDates
+                  openingDate={openingDate}
+                  closingDate={closingDate}
+                />
+                {location && (
+                  <>
+                    {'. '}
+                    {location.getIn(['cities', 0])}
+                    {', '}
```

Comment:
how about when it doesn't have a `city` or `country`?

Reply:
maybe it makes sense to move this another component `ConferenceAddress` and implement all this logic there and test separately.

Reply:
I think all the records must have `city` and `country`, right? At least according to the required fields in the submission form task.
Also, now we know that `cnum` is required, so it's not really necessary to check that in line 91 I guess.